lib/Gateway.js

```diff
@@ -605,6 +605,33 @@ function getMappedValuesTemplate (modeMap, defaultValue) {
   )}}[value_json.value] | default('${defaultValue}') }}`
 }
 
+/**
+ * Calculate the correct template string to use for templates with state
+ * list based on gateway settings and mapped mode values
+ *
+ * @param {Object} state The object list which is translated to map
+ * @param {String} defaultValueKey The key to use for default value
+ * @returns {String} The template to use for the template
+ */
+function getMappedStateTemplate (state, defaultValueKey) {
+  const map = []
+  let defaultValue = 'value_json.value'
+  for (const listKey in state) {
+    map.push(
+      `${
+        typeof state[listKey].value === 'number'
+          ? state[listKey].value
+          : '"' + state[listKey].value + '"'
+      }: "${state[listKey].text}"`
+    )
+    if (state[listKey].value === defaultValueKey) {
+      defaultValue = `'${state[listKey].text}'`
+    }
+  }
+
+  return `{{ {${map.join(',')}}[value_json.value] | default(${defaultValue}) }}`
```

Comment:
Shouldn't this be without "default()"?

Reply:
No. if this is not having a default(), it returns no value.. aka empty at all, where in this case if there is no defined default in payload... unmapped value will return the number or string.

Basically the default here is if the map translation fails, what to show